rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LRARecord.java

```diff
@@ -364,12 +372,27 @@ private int tryDoEnd(boolean compensate) {
             }
         }
 
-        if (httpStatus == Response.Status.OK.getStatusCode() && responseData != null) {
-            // see if any content contains a failed status
-            if (compensate && LRAStatus.FailedToCancel.name().equals(responseData)) {
-                return reportFailure(true, endPath.toASCIIString());
-            } else if (!compensate && LRAStatus.FailedToClose.name().equals(responseData)) {
-                return reportFailure(false, endPath.toASCIIString());
+        if (responseData != null) {
+            String failureReason = null;
+
+            if (httpStatus == Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()) {
+                // the body should contain a valid ParticipantStatus
+                try {
+                    failureReason = ParticipantStatus.valueOf(responseData).name();
+                } catch (IllegalArgumentException ignore) {
+                    // ignore the body and let recovery discover the status of the participant
+                }
+            } else if (httpStatus == Response.Status.OK.getStatusCode()) {
+                // see if any content contains a failed status
+                if (compensate && LRAStatus.FailedToCancel.name().equals(responseData)) {
```

Comment:
shouldn't this be ParticipantStatus.FailedToCompensate? Similarly for FailedToComplete.

Reply:
Ah, I think that might well be right actually. @mmusgrov, please can you comment? I will move it to request changes to block the merge until we get an answer

Reply:
> LGTM - but I think we need a test beyond the TCK [one](https://github.com/eclipse/microprofile-lra/pull/174/files#diff-3eeea840a3958c8e4d50c676076e87a5R259). For example crash recovery one? It would be OK to add the Jira to do this work after IMO

I created a TCK issue for it https://github.com/eclipse/microprofile-lra/issues/190

Reply:
The code that used the wrong status is now dead code since we no longer use the body in 200 responses. I have just removed it.